Makefile

```diff
@@ -31,7 +31,27 @@ ifeq ($(GOOS),windows)
 	IS_EXE := .exe
 endif
 MINISHIFT_BINARY ?= $(GOPATH)/bin/minishift$(IS_EXE)
-GODOG_OPTS ?= ""
+ifeq ($(filter-out integration,$(MAKECMDGOALS)),all)
+    GODOG_OPTS ?= -tags basic,cmd-openshift,command,coolstore,flags,openshift-version,proxy -format pretty
+endif
+ifeq ($(findstring .feature,$(filter-out integration,$(MAKECMDGOALS))), .feature)
+    ARGUMENTS = $(subst " ", ,$(filter-out integration,$(MAKECMDGOALS)))
+    ARGUMENT1 = $(word 1, $(ARGUMENTS))
+    ARGUMENT2 = $(word 2, $(ARGUMENTS))
+    ifeq ($(ARGUMENT2),all)
+        GODOG_OPTS ?= -paths $(ARGUMENT1) -tags basic,cmd-openshift,command,coolstore,flags,openshift-version,proxy -format pretty
+    else ifeq ($(ARGUMENT1),$(lastword $(MAKECMDGOALS)))
+	    GODOG_OPTS ?= -paths $(ARGUMENT1)
+    else
+	    GODOG_OPTS ?= -paths $(ARGUMENT1) -tags $(ARGUMENT2) -format pretty
+    endif
+else
+    ifeq (integration,$(lastword $(MAKECMDGOALS)))
+        GODOG_OPTS ?= ""
+    else
+    	GODOG_OPTS ?= -tags $(filter-out integration,$(MAKECMDGOALS)) -format pretty
+    endif
+endif
```

Comment:
@amitkrout This looks complicated. I am sure if it is doing the right thing. Can you please explain the logic you are using?

Reply:
@LalatenduMohanty Even i find it more complicated while writing it ;)

Reply:
@LalatenduMohanty commented on each condition